### DISCUSSION TITLE
What's the fastest way to stack pytrees?

### DISCUSSION THREAD

**Discussion by NeilGirdhar** [jax-ml/jax#9855] [Q&A] [answered]
I was forced to convert a `jax.lax.scan` into an ordinary loop that produces a list of pytrees.  I want to stack them, but it is incredibly slow:
```python
        def stack(*elements: Array) -> Array:
            return jnp.stack(elements)

        with Timer() as timer:
            trajectory = tree_map(stack, *results)  # results has 5000 trees in it
        log.info(f"Stacking took {timer.elapsed_str(3)}")  # Stacking took 71.2 s!!!
```
What's the fastest way to accomplish this in Jax?

**Comment by NeilGirdhar** [comment, marked as answer]
Switching to `np.stack` brought the runtime down to 2s!

**Reply by YouJiacheng** [reply]
But `np.stack` cannot `jit`?

**Reply by NeilGirdhar** [reply]
@YouJiacheng True, but if I could use the JIT, I would also be using `lax.scan`.
